plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from pathlib import Path
+from typing import Any
+
+import pytest
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.test_utils.test_utils import TSweepRunner, chdir_plugin_root, get_run_output
+from omegaconf import DictConfig, OmegaConf
+from optuna.distributions import (
+    CategoricalDistribution,
+    DiscreteUniformDistribution,
+    IntLogUniformDistribution,
+    IntUniformDistribution,
+    LogUniformDistribution,
+    UniformDistribution,
+)
+
+from hydra_plugins.hydra_optuna_sweeper import _impl
+from hydra_plugins.hydra_optuna_sweeper.optuna_sweeper import OptunaSweeper
+
+chdir_plugin_root()
+
+
+def test_discovery() -> None:
+    assert OptunaSweeper.__name__ in [
+        x.__name__ for x in Plugins.instance().discover(Sweeper)
+    ]
+
+
+def assert_optuna_distribution_equals(expected: Any, actual: Any) -> None:
+    assert type(expected) == type(actual)
```

Comment:
Is it possible to directly compare the two objects instead of comparing them field by field?

Reply:
Thank you for your suggestion. You're absolutely right. I removed `assert_optuna_distribution_equals` and used `assert expected == actual` instead in the commit https://github.com/facebookresearch/hydra/pull/1132/commits/18714b5c339c2af95972d1672470e0dfd33faf30. This change really simplified the test code.